server/pulp/server/webservices/views/repositories.py

```diff
@@ -0,0 +1,1214 @@
+        raise pulp_exceptions.OperationPostponed(async_result)
+
+
+class RepoPublish(View):
+    """
+    View for publishing a repository.
+    """
+
+    @auth_required(authorization.EXECUTE)
+    @json_body_allow_empty
+    def post(self, request, repo_id):
+        """
+        Dispatch a task to publish a repository.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_id: id of the repository to publish
+        :type  repo_id: str
+
+        :raises pulp_exceptions.OperationPostponed: dispatch a publish repo task
+        """
+
+        # validation
+        manager = manager_factory.repo_query_manager()
+        manager.get_repository(repo_id)
+
+        distributor_id = request.body_as_json.get('id', None)
+        overrides = request.body_as_json.get('override_config', None)
+        async_result = repo_tasks.publish(repo_id, distributor_id, overrides)
+        raise pulp_exceptions.OperationPostponed(async_result)
+
+
+class RepoAssociate(View):
+    """
+    View to copy units between repositories.
+    """
+
+    @auth_required(authorization.UPDATE)
+    @json_body_required
+    def post(self, request, dest_repo_id):
+        """
+        Associate units matching the criteria into the given repository
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param dest_repo_id: id of the repository content will be copied to
+        :type  dest_repo_id: str
+
+        :raises pulp_exceptions.MissingValue: if required param source_repo_id is not passed
+        :raises pulp_exceptions.InvalidValue: if source_repo_id is not found or if criteria
+                                              params cannot be parsed
+        :raises pulp_exceptions.OperationPostponed: dispatch a publish repo task
+        :raises pulp_exceptions.PulpDataException: if there is a problem parsing the criteria object
+        """
+
+        # Validate existence of dest_repo_id, can raise a MissingResource
+        repo_query_manager = manager_factory.repo_query_manager()
+        repo_query_manager.get_repository(dest_repo_id)
+
+        criteria_body = request.body_as_json.get('criteria', None)
+        overrides = request.body_as_json.get('override_config', None)
+        source_repo_id = request.body_as_json.get('source_repo_id', None)
+        if source_repo_id is None:
+            raise pulp_exceptions.MissingValue(['source_repo_id'])
+
+        # Catch MissingResource because this is body data, raise 400 rather than 404
+        try:
+            repo_query_manager.get_repository(source_repo_id)
+        except pulp_exceptions.MissingResource:
+            raise pulp_exceptions.InvalidValue(['source_repo_id'])
+
+        if criteria_body:
+            try:
+                criteria = UnitAssociationCriteria.from_client_input(criteria_body)
+            except pulp_exceptions.InvalidValue, e:
```

Comment:
That code was eating all exceptions and returning an unhelpful one. 
https://pulp.plan.io/issues/804